src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadListWidget.java

```diff
@@ -78,29 +75,45 @@ public void configure(String forumId, Boolean isCurrentUserModerator,
 		} else {
 			this.filter = DEFAULT_FILTER;
 		}
+		loadMoreWidgetContainer.configure(new Callback() {
+			@Override
+			public void invoke() {
+				loadMore();
+			}
+		});
 		loadMore();
 		DiscussionThreadCountAlert threadCountAlert = ginInjector.getDiscussionThreadCountAlert();
 		view.setThreadCountAlert(threadCountAlert.asWidget());
 		threadCountAlert.configure(forumId);
 	}
 
-	public void clear() {
-		threadsContainer.clear();
-	}
-	
-	public void setThreadIdClickedCallback(CallbackP<String> threadIdClickedCallback) {
-		this.threadIdClickedCallback = threadIdClickedCallback;
+	public void configure(String entityId, CallbackP<Boolean> emptyListCallback,
+			DiscussionFilter filter) {
+		clear();
+		this.emptyListCallback = emptyListCallback;
+		offset = 0L;
+		this.entityId = entityId;
+		if (filter != null) {
+			this.filter = filter;
+		} else {
+			this.filter = DEFAULT_FILTER;
+		}
+		loadMoreWidgetContainer.configure(new Callback() {
+			@Override
+			public void invoke() {
+				loadMoreThreadsForEntity();
```

Comment:
very clever.  you define a different "load more" callback depending on how the widget is configured.